Clean up validation error messages

emanchado opened the pull request:
In validation code we often throw exceptions inside exception
handlers, which makes Python report the exception like a problem while
handling the original exception. That's not good, first because it
looks like a bug in our code, and second because the validation error
output on the console becomes pretty big and it's very confusing to
read. So, explicitly clear up all references to the exception
"context" (by using "from None" at the end of the raise statement) for
all exceptions thrown inside a exception handler.

Ref. FASTMOVING-1456

spbnick reviewed (changes requested):
Yes, this was half-backed, and I simply didn't have time to go and improve it.
However, instead of fighting the Python's exception chaining, why not use it instead?
Just drop the `kpet.schema.Invalid.__str__()` method altogether, catch all exceptions in `kpet.main()` and print them using the method in the former, then exit with `1`.

emanchado committed:
Clean up validation error messages

Revamp the way we report validation exceptions. In validation code we
often throw exceptions inside exception handlers, and because of the
way we were reporting them, the error messages the user saw had lots
of repeated bits, to the point of making the output unreadable (and
maybe even making the user think that there's a bug in
kpet!). Instead, show a more helpful error message for the user which
doesn't repeat information.

A couple of tests had to be tweaked slightly to adapt to the
differences in error reporting.

Ref. FASTMOVING-1456

spbnick reviewed (approved):
LGTM! Thanks, Esteban :)

spbnick reviewed (approved):
Looks good! Thanks a lot, Esteban :)!
